### PR TITLE
Use IOT sockets instead of lwip directly

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
WIP but please review to help me catch bugs

### CODE REVIEW

File: src/inet/inet.gni [hunk 1]
```diff
@@ -27,7 +27,9 @@ declare_args() {
   chip_inet_config_enable_tcp_endpoint = true
 
   # Inet implementation type.
-  if (chip_system_config_use_open_thread_inet_endpoints) {
+  if (current_os == "cmsis-rtos") {
```
Comment (ATmobica) [inline src/inet/inet.gni:30]:
```suggestion
  if (chip_system_config_use_iot_socket) {
```

File: src/system/system.gni [hunk 2]
```diff
@@ -44,8 +46,10 @@ declare_args() {
 
 declare_args() {
   # Event loop type.
-  if (chip_system_config_use_lwip ||
-      chip_system_config_use_open_thread_inet_endpoints) {
+  if (current_os == "cmsis-rtos") {
```
Comment (ATmobica) [inline src/system/system.gni:49]:
```suggestion
  if (chip_system_config_use_iot_socket) {
```

File: src/system/SystemLayerImplOpenIoTSDK.h [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    /* Return true if any of the bits are set in the any of the result masks */
+    bool IsResultSelectMaskSet()
+    {
+        uint8_t * current = (uint8_t *) GetSelectMask(ReadMaskOut);
+        uint8_t * end     = (uint8_t *) GetSelectMask(SelectMaskTypeMax);
+        while (current != end)
+        {
+            if (*current)
+            {
+                a = 1;
```
Comment (ATmobica) [inline src/system/SystemLayerImplOpenIoTSDK.h:99]: What is that `a`?

File: config/openiotsdk/chip-gn/args.gni [hunk 4]
```diff
@@ -27,6 +27,7 @@ chip_config_network_layer_ble = false
 chip_system_config_use_lwip = true
 lwip_platform = "external"
 chip_system_config_use_sockets = false
+chip_system_config_use_iot_socket = true
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply config/openiotsdk/chip-gn/args.gni:30]: the API is called iot_socket

File: config/openiotsdk/lwip/user_lwipopts.h [hunk 5]
```diff
@@ -26,9 +26,13 @@
 #define MEM_LIBC_MALLOC (1)
 #define MEM_USE_POOLS (0)
 
-#ifdef LWIP_DEBUG
+#define IP_SOF_BROADCAST 1
+#define IP_SOF_BROADCAST_RECV 1
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply config/openiotsdk/lwip/user_lwipopts.h:30]: to receive the pktinfo

File: src/system/SystemLayerImplOpenIoTSDK.h (hunk 3 above)
Comment (paul-szczepanek-arm) [reply src/system/SystemLayerImplOpenIoTSDK.h]: sorry, that was accidental debug commit, already removed

File: config/openiotsdk/lwip/user_lwipopts.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (ATmobica) [reply config/openiotsdk/lwip/user_lwipopts.h:30]: We received it with LwIP directly, why IoT socket required these? Are these due to the IoT socket implementation?

File: src/system/SystemLayerImplOpenIoTSDK.cpp [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+                }
+                break;
+            }
+        }
+    }
+
+    mSelectMutex.Unlock();
+
+    mSignalMutex.Lock();
+    osSemaphoreAcquire(mSignalSem, osWaitForever);
```
Comment (paul-szczepanek-arm) [inline src/system/SystemLayerImplOpenIoTSDK.cpp:355]: this is a workaround the LWIP bug where select does not wake on socket close resulting in assert if we open another socket that lwip assigns the reused ID which we are already waiting on

Comment (paul-szczepanek-arm) [reply src/system/SystemLayerImplOpenIoTSDK.cpp:355]: @pan-

**Pull request closed without merging**